File: Packages/CaptureKit/Sources/CaptureKit/OverlayPositioner.swift
import CoreGraphics

public enum OverlayPositioner {
    /// Returns the Cocoa (bottom-left origin) origin for an overlay window in a screen corner.
    public static func origin(corner: OverlayCorner, overlaySize: CGSize,
                              screenFrame: CGRect, margin: CGFloat) -> CGPoint {
        let leftX = screenFrame.minX + margin
        let rightX = screenFrame.maxX - overlaySize.width - margin
        let bottomY = screenFrame.minY + margin
        let topY = screenFrame.maxY - overlaySize.height - margin
        switch corner {
        case .topLeft:     return CGPoint(x: leftX,  y: topY)
        case .topRight:    return CGPoint(x: rightX, y: topY)
        case .bottomLeft:  return CGPoint(x: leftX,  y: bottomY)
        case .bottomRight: return CGPoint(x: rightX, y: bottomY)
        }
    }

    /// Origin for the overlay at stack position `index` (0 = at the corner;
    /// higher indexes step away from the screen edge so overlays pile up
    /// one over the other with `spacing` between them).
    public static func stackedOrigin(corner: OverlayCorner, overlaySize: CGSize,
                                     screenFrame: CGRect, margin: CGFloat,
                                     index: Int, spacing: CGFloat = 12) -> CGPoint {
        var o = origin(corner: corner, overlaySize: overlaySize,
                       screenFrame: screenFrame, margin: margin)
        let offset = CGFloat(index) * (overlaySize.height + spacing)
        switch corner {
        case .bottomLeft, .bottomRight: o.y += offset   // stack upward
        case .topLeft, .topRight:       o.y -= offset   // stack downward
        }
        return o
    }
}

File: windows/src/BetterScreenshot.Capture/OverlayPositioner.cs
using BetterScreenshot.Core;

namespace BetterScreenshot.Capture;

/// <summary>
/// Computes the top-left origin of an overlay window pinned to a screen corner, and stacked positions for
/// the quick-access stack. Top-left origin (Windows): bottom corners stack upward, top corners stack downward.
/// </summary>
public static class OverlayPositioner
{
    public static PxPoint Origin(Corner corner, PxSize overlay, PxRect screen, double margin)
    {
        double left = screen.X + margin;
        double right = screen.Right - overlay.Width - margin;
        double top = screen.Y + margin;
        double bottom = screen.Bottom - overlay.Height - margin;
        return corner switch
        {
            Corner.TopLeft => new PxPoint(left, top),
            Corner.TopRight => new PxPoint(right, top),
            Corner.BottomLeft => new PxPoint(left, bottom),
            Corner.BottomRight => new PxPoint(right, bottom),
            _ => new PxPoint(left, top),
        };
    }

    public static PxPoint StackedOrigin(Corner corner, PxSize overlay, PxRect screen, double margin, int index, double spacing = 12)
    {
        var baseOrigin = Origin(corner, overlay, screen, margin);
        double offset = index * (overlay.Height + spacing);
        bool isBottom = corner is Corner.BottomLeft or Corner.BottomRight;
        double y = isBottom ? baseOrigin.Y - offset : baseOrigin.Y + offset;
        return new PxPoint(baseOrigin.X, y);
    }
}
